[cutlass backend] remove triton from most tests and add an integration test

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #147485
* #147173
* __->__ #147169

Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] remove triton from most tests and add an integration test

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented on test/inductor/test_cutlass_backend.py:
```diff
@@ -128,32 +130,6 @@ def mm(a, b):
 
-    @unittest.skipIf(not SM90OrLater, "need sm_90")
-    @mock.patch.dict(os.environ, {"PATH": _get_path_without_sccache()})
-    def test_max_autotune_precompile(self):
```
what's the reason for removing this test?

henrylhtsang replied:
@ColinPeppler 
1. I am not sure what this test is testing. 
2. It would actually fail if you make fallback_to_aten False.
3. I added the integration test to replace it.

ColinPeppler replied:
it's supposed to be testing autotune in subproc, right?

is that currently broken?

henrylhtsang replied:
@ColinPeppler the failure is actually pretty interesting. I think it was failing due to alignment, since 100 and 10 are not divisible by 8.

henrylhtsang replied:
> it's supposed to be testing autotune in subproc, right?
> 
> is that currently broken?

I plan on separating the subproc tests. The subproc tests are broken for addmm, but fine for the others 
https://github.com/pytorch/pytorch/pull/147173/files

ColinPeppler commented on test/inductor/test_cutlass_backend.py:
```diff
@@ -480,10 +457,9 @@ def mm(a, b, bias):
     @mock.patch.dict(os.environ, {"PATH": _get_path_without_sccache()})
     def test_max_autotune_cutlass_backend_addmm(
-        self, dynamic, max_autotune_gemm_backends
+        self, dynamic: bool = False, max_autotune_gemm_backends: str = "CUTLASS"
```
nit: just keep as `dynamic: bool`?

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler reviewed (approved):
can you pls confirm the failed test is irrelevant?

also consider rebase to `origin/viable/strict` or comment `@pytorchbot rebase` to make sure your base is healthy

henrylhtsang commented:
> can you pls confirm the failed test is irrelevant?
> 
> also consider rebase to `origin/viable/strict` or comment `@pytorchbot rebase` to make sure your base is healthy

let me rerun the job. I don't want to reset the ci

henrylhtsang commented:
@pytorchbot merge -i

henrylhtsang committed:
Update on "[cutlass backend] remove triton from most tests and add an integration test"


Removing aten and triton from the list of backends for the tests that have it. Instead, add a small integration test to make sure autotuning works fine.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
